### DISCUSSION TITLE
How to host carbon design system locally for offline use

### DISCUSSION THREAD

**Discussion by ziggy981** [carbon-design-system/carbon#12075] [Help] [answered]
Hi,

I'm new to React and Carbon. I'd like to host the whole CarbonDesignSystem site locally. Can anyone please provide required steps after cloning the repo.

Thanks,

**Comment by J4cquelineJones** [comment]
Hi @ziggy981,

1. Once you have cloned the repo to your local system, inside your IDE or an open terminal/cmd window you will need to `cd` into the root directory of the cloned repo. 

2. The next command will be `npm i` to generate a package-lock.json file and install all the node_modules required for the application you're going to build. 

3. Once they're installed/completed use the `npm start `command to run the application in your local development server. 

If successful, you should see your opening screen in the browser.

Hope that helps :0)

**Reply by ziggy981** [reply]
Hi @J4cquelineJones , thanks for the steps. However on running npm i i'm getting dependency conflicts. Shouldn't these already be configured properly in the package.json without such versioning conflicts?

![image](https://user-images.githubusercontent.com/44898569/191204391-fdc8b30c-8c8e-4c77-826e-049d873d9893.png)

**Reply by J4cquelineJones** [reply]
@ziggy981 I can see in your screenshot that it's looking for a specific version of `react` which I am assuming you're using the latest `18.0.0` version however if you downgrade `react` to `^17.0.0` it should then run.

You can do this by using the command `npm install react@17.0.0`.

If you're still getting the same error after doing that command use this command to override `npm i --legacy-peer-deps`

**Comment by tay1orjones** [comment, marked as answer]
Hey @ziggy981 👋 

To stand up the storybook (react.carbondesignsystem.com) locally, follow the steps here in our [contributing.md](https://github.com/carbon-design-system/carbon/blob/main/.github/CONTRIBUTING.md#5-build-and-start-the-development-server)

If you're looking for carbondesignsystem.com, that is a separate repo at https://github.com/carbon-design-system/carbon-website
